@re-/model/demos/generateCode/mapTSfiles.ts

```diff
@@ -1,56 +1,133 @@
-import { fromHere } from "@re-/node"
-import { Project, Statement, ts } from "ts-morph"
+import { Project, SourceFile } from "ts-morph"
+import { Stack } from "../stack.js"
+import { addBlockToMaps, getComment } from "./helpers.js"
 import { regex } from "./regex.js"
 
-const { tagMatch, blockCommentMatch, commentMatch } = regex
+const {
+    blockEndGrab,
+    blockOrStatementGrab,
+    nameGrab,
+    includeGrab,
+    excludeGrab,
+    fileGrab,
+    statementGrab,
+    blockGrab
+} = regex
 
-const getComment = (text: string, range: any) =>
-    text.slice(range.getPos(), range.getEnd())
+const codeDestinations = ["stackBlitz", "tests", "docs"]
 
-const stripCodeTag = (codeTag: string) =>
-    codeTag.replace("@re_place", "").trim()
+const breakdownGrabComment = (comment: string, name?: string) => {
+    const NAME_TAG = "-name:"
+    let optionsDestinations: string[] = []
+    let strippedComment: string
 
-const addFileContentsToMap = (
-    fileText: string,
-    astStatements: Statement<ts.Statement>[],
-    map: Record<string, string>
-) => {
-    for (const content of astStatements) {
-        const range = content.getLeadingCommentRanges()
-        if (range.length) {
-            const foundMatchingComment = getComment(fileText, range[0]).match(
-                tagMatch
+    const commentHasNameLine = comment.match(nameGrab)
+    const nameLine =
+        name ||
+        (!!commentHasNameLine
+            ? commentHasNameLine[0].replace(NAME_TAG, "").trim()
+            : `unnamedGrab${Math.floor(Math.random() * 100000)}`)
+
+    strippedComment = comment.replace(nameGrab, "").trim()
+
+    const endCommentMatch = comment.match(blockEndGrab)
+    if (!!endCommentMatch) {
+        strippedComment = strippedComment.replace(endCommentMatch[0], "")
+    }
+    const includeMatch = strippedComment?.match(includeGrab)
+    const excludeMatch = strippedComment?.match(excludeGrab)
+
+    try {
+        if (includeMatch && excludeMatch) {
+            throw new Error(
+                "Include and Exclude in the same grab comment is not supported."
             )
-            if (
-                foundMatchingComment &&
-                foundMatchingComment[0].split(" ").length > 2
-            ) {
-                const tag = stripCodeTag(foundMatchingComment[0])
-                map[tag] = content.getText()
-            }
         }
+        if (includeMatch) {
+            optionsDestinations = codeDestinations.filter((option) =>
+                strippedComment.toLowerCase().includes(option.toLowerCase())
+            )
+        } else if (excludeMatch) {
+            optionsDestinations = codeDestinations.filter(
+                (option) =>
+                    !strippedComment
+                        .toLowerCase()
+                        .includes(option.toLowerCase())
+            )
+        } else {
+            optionsDestinations = codeDestinations
+        }
+    } catch (e: any) {
+        console.error(e.message)
```

Comment:
Is there a reason we don't want to throw here?

`console.error` will only print the error message, but I think if something went wrong in doc generation we'd just want to stop what we're doing and report the error to avoid anything unpredictable happening. If that's the case, we can just remove the `try` block wrapping this logic.

Also, in general it's probably safer to avoid accessing e.message in case what was thrown doesn't have an error property. Converting the thrown value to a string will be more reliable, i.e. `String(e)`, which `console` just does automatically for you, so in this case `console.error(e)` would be fine.